opta/core/aws.py

```diff
@@ -167,6 +162,33 @@ def force_delete_terraform_lock_id(self) -> None:
             Key={"LockID": {"S": f"{bucket}/{self.layer.name}"}},
         )
 
+    def get_all_remote_configs(self) -> Dict[str, Dict[str, "StructuredConfig"]]:
+        prefix = "opta_config/"
+        s3 = boto3.client("s3")
+        remote_configs = {}
+        for aws_bucket in self._get_opta_bucket():
+            configs = {}
+            response = s3.list_objects(Bucket=aws_bucket, Prefix=prefix, Delimiter="/")
+            if "Contents" in response:
+                for data in response["Contents"]:
+                    structured_config = self._download_remote_config(
+                        s3, aws_bucket, data["Key"]
+                    )
+                    if structured_config:
+                        configs[data["Key"][len(prefix) :]] = structured_config
+                remote_configs[aws_bucket] = configs
+        return remote_configs
+
+    @staticmethod
+    def _get_opta_bucket() -> List[str]:
```

Comment:
if more than one object can be returned let's use plural `_get_opta_buckets`
for example see the naming for the official api below `s3.list_buckets()`